### ISSUE TITLE
Flux doesn't work with K8s 1.7, kubectl incompatibility?

### ISSUE THREAD

**Issue by philwinder** [weaveworks/flux#677]
This looks like the error received when using a kubectl that is older than the k8s cluster.

```
Release job submitted, ID 4a24bdd7-d907-9b71-0076-74650969304b
Status: Failed: default/affect-detection-predict-94: applying definition to default/affect-detection-predict-94: running kubectl: error: group map[policy:0xc8203af030 rbac.authorization.k8s.io:0xc8204b4000 authentication.k8s.io:0xc8200aa000 authorization.k8s.io:0xc8203aee00 componentconfig:0xc8203aef50 apps:0xc8203aed90 autoscaling:0xc8203aee70 batch:0xc8203aeee0 extensions:0xc8203aefc0 federation:0xc8203aecb0 :0xc8203aed20] is already registered


1) Queued.
2) Calculating updates for release.
3) Cloning git repository.
4) Finding defined services.
5) Found service default/affect-detection-predict-94
6) ignored service default/affect-router as it is not running in cluster
7) Found 1 services.
8) Looking up images.
9) Will update default/affect-detection-predict-94 container affect-detection-predict-94: gcr.io/strange-cosmos-822/affect-detection:0.0.10 -> 0.0.13
10) Pushing changes.
11) Applying changes.
12) Sending notifications.
13) Failed: default/affect-detection-predict-94: applying definition to default/affect-detection-predict-94: running kubectl: error: group map[policy:0xc8203af030 rbac.authorization.k8s.io:0xc8204b4000 authentication.k8s.io:0xc8200aa000 authorization.k8s.io:0xc8203aee00 componentconfig:0xc8203aef50 apps:0xc8203aed90 autoscaling:0xc8203aee70 batch:0xc8203aeee0 extensions:0xc8203aefc0 federation:0xc8203aecb0 :0xc8203aed20] is already registered

Failure
```

E.g. 
https://github.com/kubernetes/kubernetes/issues/39494  https://github.com/ContainerSolutions/registry-tooling/issues/7 https://github.com/kubernetes/kubernetes/issues/43778

**Comment by squaremo** [comment]
I've created an image based on `quay.io/weaveworks/fluxd:0.3.0` but with kubectl v1.7.2, and pushed it to `quay.io/squaremo/fluxd:k8s-1.7`. Replacing the fluxd image with that will tell us whether the Kubernetes version is the problem.

**Comment by q-michelle** [comment]
Thanks for creating the image so quickly. I couldn't try it because I got a Permission Denied trying to pull it, but while I was editing the flux-agent yaml I noticed that it was originally set to fluxd:0.2.0. I updated it to fluxd:0.3.0 and the error was resolved. I copied the setup text from the weave cloud config page and an old definition was pulled because the v1.7 URL redirects to v1.6:
```
Michelles-Q-15:bin michelle$ curl "https://cloud.weave.works/k8s/flux.yaml?t=REDACTED&k8s-version=$(kubectl version | base64 | tr -d '\n')"
code: Redirect
message: >-
  Moved permanently to `/k8s/v1.7/flux.yaml`, assuming Kubernetes version is
  "v1.7" or older, old location (`/k8s/flux.yaml`) will be deprecated soon. If
  you are using `curl`, you can pass `--location`, but it is best to just use
  the new URL.
Michelles-Q-15:bin michelle$ curl "https://cloud.weave.works/k8s/v1.7/flux.yaml?t=REDACTED&k8s-version=$(kubectl version | base64 | tr -d '\n')"
code: Redirect
message: >-
  Moved permanently to `/k8s/v1.6/flux.yaml`, assuming Kubernetes version is
  "v1.6" or older, old location (`/k8s/v1.7/flux.yaml`) will be deprecated soon.
  If you are using `curl`, you can pass `--location`, but it is best to just use
  the new URL.
```

In short, I think we can close this issue since fluxd:0.3.0 works with k8s v1.7.1, but the URL redirect should be fixed.

**Comment by squaremo** [comment]
> I couldn't try it because I got a Permission Denied

Doh -- I didn't realise that repo was private and ought to have checked, sorry about that.

> I noticed that it was originally set to fluxd:0.2.0. I updated it to fluxd:0.3.0 and the error was resolved

Ahh right, so fluxd had been silently downgraded to 0.2.0 somewhere along the line, because our launch-generator supplies it in config. Good sleuthing :)

I'm going to close this issue, since it wasn't quite what we thought it was. We'll still need to get the automatically-generated config sorted out though.
